encoder/vaapiencoder_base.h

```diff
@@ -178,7 +178,7 @@ class VaapiEncoderBase : public IVideoEncoder {
     OutputQueue m_output;
 
     bool updateMaxOutputBufferCount() {
-        if (m_maxOutputBuffer < m_videoParamCommon.leastInputCount + 3)
+        if ((m_videoParamCommon.leastInputCount + 3 >= 0) && (m_maxOutputBuffer < (uint32_t)(m_videoParamCommon.leastInputCount + 3)))
```

Comment:
m_maxOutputBuffer is type of uint32_t, so its value will bigger than -1. 
if "m_videoParamCommon.leastInputCount + 3" is negative, "(uint32_t)(m_videoParamCommon.leastInputCount + 3)" may be bigger than m_maxOutputBuffer.